docs/how-to/builtins.rst

```diff
@@ -1,104 +1,138 @@
 Implementing Python Built-ins in JavaScript
 ===========================================
 
+Python's builtins give Python its trademark magical feel. If you're new to Python, read up on them here: https://docs.python.org/3/library/functions.html
+
+Most builtins have already been added to the project, but many do not follow the spec outlined below.
+Some may not handle certain types of inputs correctly. In addition, new builtins may arrive with the
+latest and greatest Python version. This guide should serve as your field manual for adding, updating, and
+navigating our implementations.
+
+Process
+-------
+
+The first thing to do when adding anything to Batavia is to play around a bit with it in the Python REPL.
+Here's an example using ``list()``::
+
+    >> list()
+    []
+    >> list((1, 2, 3, 4))
+    [1, 2, 3, 4]
+    >> list(4)
+    Traceback (most recent call last):
+    File "<stdin>", line 1, in <module>
+    TypeError: 'int' object is not iterable
+
+Your goal is to find out how the function responds to various inputs and outputs. You may also
+want to consult the offical documentation. Once you're a little familiar, you can start to add your
+implementation to Batavia.
+
 General Structure
------------------
+*****************
 
 JavaScript versions of Python built-in functions can be found inside the ``batavia/builtins``
-directory in the Batavia code. Each built-in is placed inside its own file.
+directory in the Batavia code. Each built-in is placed inside its own file. These builtins are 
+designed to be used only inside Batavia, as such they need to ensure they are being used in
+a compatible manner.
 
-.. code-block:: javascript
+Each builtin function will receive arguments and keyword arguments and needs to handle them,
+even if the result is throwing an error. Args should be an array, and kwargs should be a 
+JavaScript object. The first thing to do is check that both were passed in.
 
-    // Example: a function that accepts exactly one argument, and no keyword arguments
+.. code-block:: javascript
 
-    var <fn> = function(<args>, <kwargs>) {
-        // These builtins are designed to be used only inside Batavia, as such they need to ensure
-        // they are being used in a compatible manner.
+    // Example: <fn> accepts exactly one argument, and no keyword arguments
```

Comment:
Since we're cleaning this up... I'm not wild about using "<fn>" as an example. It's potentially confusing, since as written, it's not valid Javascript *or* Python syntax. I'd rather use an actual concrete example, with a preface note that says "we're going to use abs() as an example here; abs() is already implemented, but the pattern will be the same for any other builtin". (Substitute abs() for whatever builtin makes sense in the content)